frontend/src/v5/ui/components/shared/appBar/appBar.component.tsx

```diff
@@ -15,31 +15,153 @@
-import { Items } from './appBar.styles';
+import ContactUsIcon from '@assets/icons/email.svg';
+import InviteAFriendIcon from '@assets/icons/add_user.svg';
+import TeamspacesIcon from '@assets/icons/teamspaces.svg';
+import VisualSettingsIcon from '@assets/icons/settings.svg';
+import SupportCentreIcon from '@assets/icons/question_mark.svg';
+import { CurrentUserHooksSelectors } from '@/v5/services/selectorsHooks/currentUserSelectors.hooks';
+import { AvatarButton } from '@controls/avatarButton';
+import {
+	Items,
+	UserMenu,
+	Section,
+	AvatarSection,
+	Popper,
+	Paper,
+	UserFullName,
+	UserUserName,
+	SignOutButton,
+	EditProfileButton,
+	Avatar,
+} from './appBar.styles';
 import { Breadcrumbs } from '../breadcrumbs';
+import { UserMenuButton } from './userMenuButton/userMenuButton.component';
 
-export const AppBar = (): JSX.Element => (
-	<MuiAppBar position="static">
-		<Items>
-			<LogoIcon />
-			<Breadcrumbs />
-		</Items>
-		<Items>
-			<CircleButton variant="contrast" aria-label="intercom">
-				<IntercomIcon />
-			</CircleButton>
-			<CircleButton variant="contrast" aria-label="notifications">
-				<NotificationsIcon />
-			</CircleButton>
-			<AvatarButton> GH </AvatarButton>
-		</Items>
-	</MuiAppBar>
+const getUserNamesInitials = ({ firstName, lastName }) => (
+	[firstName, lastName]
+		.map((name) => name.trim().charAt(0).toUpperCase())
+		.join('')
 );
+
+export const AppBar = (): JSX.Element => {
+	const { teamspace } = useParams();
+	const { url } = useRouteMatch();
+	const baseUrl = url.split('/').slice(0, 3).join('/');
+	const userIsPending = CurrentUserHooksSelectors.selectIsPending();
+	const user = CurrentUserHooksSelectors.selectCurrentUser();
+
+	const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+	const handleCloseDropdown = () => {
+		setAnchorEl(null);
+	};
+
+	const handleClickDropdown = (event: MouseEvent<HTMLButtonElement>) => {
+		event.stopPropagation();
+		setAnchorEl(event.currentTarget);
+	};
+
+	return (
+		<>
+			<MuiAppBar position="static">
+				<Items>
+					<LogoIcon />
+					<Breadcrumbs />
+				</Items>
+				<Items>
+					<CircleButton variant="contrast" aria-label="intercom">
+						<IntercomIcon />
+					</CircleButton>
+					<CircleButton variant="contrast" aria-label="notifications">
+						<NotificationsIcon />
+					</CircleButton>
+					<AvatarButton
+						onClick={handleClickDropdown}
+						src={user.hasAvatar ? user.avatarUrl : null}
+					>
+						{!userIsPending && getUserNamesInitials(user)}
+					</AvatarButton>
+				</Items>
+			</MuiAppBar>
+			<Popper
+				open={Boolean(anchorEl)}
+				anchorEl={anchorEl}
+				transition
+				disablePortal
+				placement="bottom-end"
+			>
+				{({ TransitionProps, placement }) => (
+					<Grow
+						{...TransitionProps}
+						style={{ transformOrigin: placement === 'bottom-end' ? 'center top' : 'center bottom' }}
+					>
+						<Paper>
+							<ClickAwayListener onClickAway={handleCloseDropdown}>
+								<UserMenu>
```

Comment:
We could make the whole dropdown list a separate `<UserMenu>` component.

Reply:
done